Marlin/Marlin_main.cpp

```diff
@@ -3298,6 +3410,45 @@ inline void gcode_G28() {
         } //xProbe
       } //yProbe
 
+      #if ENABLED(PROBE_FAIL_PANIC)
+        if(reprobe_attempts == NUM_ATTEMPTS && probePointCounter == -1)
+        {
+          if(!IS_SD_PRINTING)
+            probe_fail = true;
+            disable_all_heaters();
+          #if ENABLED(THERMAL_RUNAWAY_PROTECTION_PERIOD)
+            for(int i=0; i<EXTRUDERS; i++) target_temp_reached[i] = false;
```

Comment:
A minor syntax point. The Marlin code prefers spaces around operators like ` - `, ` + `, ` > `, ` < `, etc.